chore(repo): move library into cadc-gmui-server/ subdir (history preserved) and add gmui application as a sibling subdir

WenbinWL opened the pull request:
This PR reorganizes the repo structure:

- Moved the existing library into `cadc-gmui-server/` subdir (history preserved).
- Added the GMUI application under `gmui/` as a sibling subdir.
- Unified `.github/workflows/ci.yml` at the repo root (builds both projects).
- No logic changes, just structural cleanup.

_Note: no secrets or registry information are added yet — this PR is purely structural. Runtime config and image publishing setup will come later._

@at88mph @pdowler — could you please confirm that this structure is what you had in mind?

Commit:
chore(repo): move library into cadc-gmui-server/ subdir (history preserved)

Commit:
feat(gmui): add gmui application snapshot under gmui/

Commit:
ci: add unified workflow for library (JDK11) and gmui (JDK17 + docker build)

Commit:
chore: add .gitignore and remove build artifacts from repo

Commit:
ci: change gmui build to use JDK11 instead of JDK17

Commit:
build: update cadc-gmui-server to Java 11

Commit:
fix: update TableWriter API calls to use OutputStream instead of Writer

- Replace direct Writer usage with OutputStream adapter in 3 resource files
- Fixes compilation error with newer CADC library versions that changed TableWriter API
- Maintains backward compatibility by converting bytes to UTF-8 strings for Writer

WenbinWL commented:
- Code Fix

Problem: The latest CADC libraries changed TableWriter.write() to accept OutputStream instead of Writer.

> Solution: Added a lightweight OutputStream adapter in three resource classes to bridge to the existing Writer.
> Affected files:
> 1. GroupAdminListResource.java
> 2. GroupMemberListResource.java
> 3. GroupListResource.java

pdowler reviewed (changes requested):
I guess more of the things in the root of the repo should not have been moved; should be easy to move them back:
- LICENSE
 - gradle wrapper (dir and the two scripts); the CI will need to refer to it as `../gradlew`

I am less sure about `settings.gradle` but I think you can remove it; we stick with maven/gradle naming conventions when possible so we only need the minimal config.

I came to dislike settings.gradle in the root of a multi-module repo because it ends up requiring that all modules have valid build files and build, which makes it hard to fix things methodically. I'm sure there are _pros_ to having it, but so far I've only experienced the _cons_ :-)

Commit:
refactor: restructure project layout as recommended

- Move LICENSE to root directory (shared across projects)
- Move gradle wrapper files to root directory for shared usage
- Delete cadc-gmui-server/settings.gradle (not needed for single module)
- Delete duplicate gmui/LICENSE file
- Update CI to use ../gradlew from submodule directories
- Follows multi-module project best practices

pdowler commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,27 @@
+on:
+  push:
+  pull_request:
+
```
if you look at other CI builds for java code, we do a single setup step and then a sequence of simple builds. Tnat ends up being more maintainable setup actions change. In this case it also makes the `gmui` build depend on a possibly updated library rather than be an independent job.

pdowler commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,27 @@
+      - uses: actions/setup-java@v4
+        with: { distribution: temurin, java-version: '11', cache: gradle }
+      - uses: gradle/actions/setup-gradle@v3
+      - run: ../gradlew -x test clean war
```
this should be `../gradlew clean build` so it runs unit tests

Commit:
refactor: cleanup based on feedback

- Remove redundant gradle wrapper from gmui/
- Clean unused test config from gmui/build.gradle
- Move README.md to root directory
- Update CI to sequential build with tests enabled

Commit:
fix: add tests and remove redundant gradle wrapper

- Add missing test dependencies to gmui/build.gradle
- Restore integration test configuration for gmui
- Remove duplicate gradle wrapper from gmui directory
- Update CI job descriptions for clarity

at88mph reviewed (approved):
I would just note that the `cadc-gmui-server` library is technical debt and should be merged into a single repo (`cadc-group-management-ui`) as sharing that library would be difficult.  Looks fine otherwise.
